### PR TITLE
Mod 2838 - Add Sample and UT for TagTechnologyMifareUltralight

### PR DESCRIPTION
**Pull request by anil-shukla-axway**
https://jira.appcelerator.org/browse/MOD-2838

### CODE REVIEW

File: ios/example/app.js [hunk 1]
```diff
@@ -1,90 +1,175 @@
...
+              logs.push('Length of response object : ' + e.responseDataLength);
+              setData(logs);
+          }
+      });
+      nfcAdapter.invalidate(sessionType);
+  });
+   if (mifare){
+  mifare.connect({mifare});
+   }else {
+       alert('No mifare tag available to disconnect');
```
Comment (fasihul-axway) [inline ios/example/app.js:119]: i think you mean connect 
```suggestion
       alert('No mifare tag available to connect');
```

File: ios/example/app.js [hunk 2]
```diff
@@ -1,90 +1,175 @@
...
+var clearLogBtn = Titanium.UI.createButton({
+  left: 50,
+  top: 150,
+  title: 'Clear Log'
+});
+
+clearLogBtn.addEventListener('click', function() {
+   var rowCount = logs.length;
+   for (var i = rowCount ; i > -1; i--) {
+       tableView.deleteRow(i);
```
Comment (fasihul-axway) [inline ios/example/app.js:136]: you are only deleting table row, you should also clear array.

File: ios/example/app.js [hunk 3]
```diff
@@ -1,90 +1,175 @@
...
+              logs.push('Error when sending command with - error code ' + e.errorCode + ' error domain: ' + e.errorDomain + ' error description ' + e.errorDescription);
+              setData(logs);
+          }else {
+              Ti.API.info('Connected tag object : ' + e.responseDataLength);
+              logs.push('Command send to Mifare tag to read page data');
+              logs.push('Length of response object : ' + e.responseDataLength);
+              setData(logs);
+          }
+      });
+      nfcAdapter.invalidate(sessionType);
```
Comment (fasihul-axway) [inline ios/example/app.js:114]: why we are invalidating session? if i am removing **invalidate** it is giving me error of timeout, do you know why?

Printing description of error:
Error Domain=NFCError Code=201 "Session timeout" UserInfo={NSLocalizedDescription=Session timeout}

**Review by fasihul-axway — CHANGES_REQUESTED** [review]
Please see my comments, also i see too many if and else, its good if we can minimise it.

File: ios/example/app.js (hunk 3 above)
Comment (anil-shukla-axway) [reply ios/example/app.js:114]: Session has begin for mifare tag reading, So after the the connection got completed I have invalidated the session.

File: ios/example/app.js [hunk 4]
```diff
@@ -1,90 +1,176 @@
...
+});
+var clearLogBtn = Titanium.UI.createButton({
+  left: 50,
+  top: 150,
+  title: 'Clear Log'
+});
+
+clearLogBtn.addEventListener('click', function() {
+   var rowCount = logs.length;
+   for (var i = rowCount ; i > -1; i--) {
```
Comment (fasihul-axway) [inline ios/example/app.js:135]: @anil-shukla-axway, i do no think any loop is required to clear the logs. You can clear logs like...

```suggestion
   logs.splice(0, logs.length);
    setData(logs);
```

**Review by fasihul-axway — COMMENTED** [review]
@anil-shukla-axway, it good if you add code to remove the added events, and see if we can minimise the if else.

File: ios/example/app.js (hunk 1 above)
Comment (anil-shukla-axway) [reply ios/example/app.js]: Done

File: ios/example/app.js (hunk 2 above)
Comment (anil-shukla-axway) [reply ios/example/app.js]: Updated the code as suggested

File: ios/example/app.js (hunk 4 above)
Comment (anil-shukla-axway) [reply ios/example/app.js]: Updated

**Comment by anil-shukla-axway** [comment]
> @anil-shukla-axway, it good if you add code to remove the added events, and see if we can minimise the if else.

I have checked the error object on each event calling, so that if any error occur, the error will get print on logs table view.

**Comment by fasihul-axway** [comment]
> > @anil-shukla-axway, it good if you add code to remove the added events, and see if we can minimise the if else.
> 
> I have checked the error object on each event calling, so that if any error occur, the error will get print on logs table view.

@anil-shukla-axway, i think the **if** and **else** statements can be reduced like 

`logs.push('didDetectTags with message' + (errorCode ? (' error code: ' + errorCode + ' error domain: ' + errorDomain + ' error description: ' + errorDescription) : ': Tag Detected'));`

**Review by fasihul-axway — APPROVED** [review]
I think **if** and **else** can be minimise like

`logs.push('didDetectTags with message' + (errorCode ? (' error code: ' + errorCode + ' error domain: ' + errorDomain + ' error description: ' + errorDescription) : ': Tag Detected'));`

But since it is not blocker, i am approving, i will recommend to use above code or something similar approach to minimise too many if and else statement.

**Review by vikas-goyal-axway — APPROVED** [review]

**Pull request merged**